WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -151,6 +151,12 @@ final class ProductFormViewModel: ProductFormViewModelProtocol {
             }
         }()
 
+        // Preview existing drafts or new products, that can be saved as a draft
+        if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.productsPreview),
+           (canSaveAsDraft() || productModel.status == .draft) {
+            buttons.insert(.preview, at: 0)
+        }
```

Comment:
Do you think we could move this logic to the switch above, or would things get more complicated? The switch is useful for us to not miss any case or combination. 

Also, do you think this is worth unit testing?

Reply:
This logic doesn't really fit in switch cases + it requires FeatureFlag check. If we add it to switch above, it'll explode into more cases and it'll be hard to understand when "preview" flow is applied.
I've updated it with comments in b6ea899, c1e812f.

Unit tests added in d69489c.